tests/test_oauth2client.py

```diff
@@ -305,6 +297,18 @@ def test_save_to_well_known_file_service_account(self):
     self.assertEqual('ABCDEF', d['private_key_id'])
     os.remove(temp_credential_file)
 
+  def test_save_well_known_file_with_non_existent_config_dir(self):
+    credential_file = datafile(
+        os.path.join('gcloud', 'application_default_credentials.json'))
+    credentials = _get_application_default_credential_from_file(
+        credential_file)
```

Comment:
well, in principle `save_to_well_known_file` could do things like validate the credential itself before looking for the file location. given that it's easy to pass a valid credential, might as well, right?